internal/proxy/reverseproxy/reverseproxy.go

```diff
@@ -48,6 +65,48 @@ func (rp *ReverseProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 
 	if rr.err != nil {
 		logger.Warnf("Error %q - code: %d", rr.err, rr.statusCode)
+
+		if rr.statusCode == http.StatusForbidden {
+			var err error
+			rp.IsValidSession, err = checkSessionValidity(req)
+			if err != nil {
+				logger.Error(err)
+			}
+		}
 		http.Redirect(rw, req, rp.RedirectURL.String(), http.StatusFound)
 	}
 }
+
+func checkSessionValidity(req *http.Request) (bool, error) {
+	requestURL := req.URL
+	cookies := req.Cookies()
+
+	jenkinsURL := fmt.Sprintf("%s://%s", requestURL.Scheme, requestURL.Host)
+
+	for _, cookie := range cookies {
```

Comment:
What is expected behavior if cookies contain more than 1 `sessions cookies`. ?

1) It should go through all session cookies and evaluate it's valid or not. If it found a valid one then it should continue the redirection normal redirection flow? 
If yes then this code would need to be fixed. At this moment it would evaluate to `false` even though cookies contains valid session.

Reply:
We have code that makes sure that we don't end up in a situation where we have two session cokkies.
https://github.com/fabric8-services/fabric8-jenkins-proxy/blob/b065e0b31d87219e076df01da1f0ff1d9b32b212/internal/proxy/ui_requests.go#L102